main.tf

```diff
@@ -188,7 +193,8 @@ module "container_definition" {
     "--providers.ecs.region=${module.this.aws_region}",
     "--providers.ecs.autodiscoverclusters=true",
     "--providers.ecs.exposedbydefault=false",
-    "--providers.ecs.defaultrule=Host(`{{ index .Labels \"Application\"}}.{{ index .Labels \"Domain\"}}`)"
+    "--providers.ecs.defaultrule=Host(`{{ index .Labels \"Application\"}}.{{ index .Labels \"Domain\"}}`)",
+    var.enable_prometheus_metrics ? "--metrics.prometheus=${var.enable_prometheus_metrics} --entryPoints.metrics.address=:9100" : null,
```

Comment:
same comment as above, please fix the same way you do above.
At the same time i would make the two arguments two separate list entries, as otherwise i would assume that this might in fact not even work due to being recognized as one argument.